src/Build.UnitTests/Evaluation/ExpanderFunction_Tests.cs

```diff
@@ -162,23 +162,26 @@ public void TryConvertToLongGivenDoubleWithLongMaxValueShouldNotThrow()
             _ = Should.NotThrow(() => Expander<IProperty, IItem>.Function<IProperty>.TryConvertToLong((double)long.MaxValue, out _));
         }
 
-        [Fact]
-        public void TryConvertToLongGivenDoubleWithLongMaxValue()
+        [WindowsFullFrameworkOnlyFact]
+        public void TryConvertToLongGivenDoubleWithLongMaxValueFramework()
```

Comment:
I don't think I understand this test change. Can you elaborate on the reasoning?

Reply:
The test previously introduced here https://github.com/dotnet/msbuild/commit/0f6ed65aea401b96a639c8b84db7d0688a48591d
Testing different behaviour on different archtitecture. 
However after moving to .net 9 the precision is not lost hence we are receveing the expected outcome of what was written: 

```
result.ShouldBeTrue();
actual.ShouldBe(longMaxValue);
```
This is true only for .NET 9 target, and the Framework left as it is.